config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/environment/FullBundleCreator.java

```diff
@@ -109,27 +109,34 @@ private String createFullBundleAsString(Map<String, String> environmentPropertie
-        Element referencesElement = getSingleChildElement(bundleElement, REFERENCES);
-        Element mappingsElement = getSingleChildElement(bundleElement, MAPPINGS);
-
-        // store Set of elements previously added so avoiding repetition in the resulting bundle
-        Set<String> addedItems = new HashSet<>();
-        Set<String> addedMappings = new HashSet<>();
-
-        // merge the deployment bundles into the environment one to get the full bundle
-        templatizedBundles.forEach(tb -> {
-            try {
-                final Element detemplatizedBundleElement = documentTools.parse(tb.getContents()).getDocumentElement();
-                copyNodes(getSingleChildElement(detemplatizedBundleElement, REFERENCES), ITEM, document, referencesElement, item -> addedItems.add(buildBundleItemKey(item)));
-                copyNodes(getSingleChildElement(detemplatizedBundleElement, MAPPINGS), MAPPING, document, mappingsElement, mapping -> {
-                    final String key = buildBundleMappingKey(mapping);
-                    return addedItems.contains(key) && addedMappings.add(key);
-                });
-            } catch (DocumentParseException e) {
-                throw new EntityBuilderException("Unable to read bundle " + tb.getName(), e);
-            }
-        });
+        //ToDo : Need to handle bundle name and version properly
+        Map<String, Element> bundleElements = bundleEntityBuilder.build(environmentBundle, EntityBuilder.BundleType.ENVIRONMENT, document, bundleFileName, "");
+        Set<Map.Entry<String, Element>> entrySet =  bundleElements.entrySet();
+        Element bundleElement = null;
+        for(Map.Entry<String, Element> entry: entrySet) {
+            // generate the environment bundle
+            bundleElement = entry.getValue();
+            Element referencesElement = getSingleChildElement(bundleElement, REFERENCES);
+            Element mappingsElement = getSingleChildElement(bundleElement, MAPPINGS);
+
+            // store Set of elements previously added so avoiding repetition in the resulting bundle
+            Set<String> addedItems = new HashSet<>();
+            Set<String> addedMappings = new HashSet<>();
+
+            // merge the deployment bundles into the environment one to get the full bundle
+            templatizedBundles.forEach(tb -> {
+                try {
+                    final Element detemplatizedBundleElement = documentTools.parse(tb.getContents()).getDocumentElement();
+                    copyNodes(getSingleChildElement(detemplatizedBundleElement, REFERENCES), ITEM, document, referencesElement, item -> addedItems.add(buildBundleItemKey(item)));
+                    copyNodes(getSingleChildElement(detemplatizedBundleElement, MAPPINGS), MAPPING, document, mappingsElement, mapping -> {
+                        final String key = buildBundleMappingKey(mapping);
+                        return addedItems.contains(key) && addedMappings.add(key);
```

Comment:
I didn't understand the logic here. Can you please add a comment explaining why `.add()` is added in the condition.

Reply:
I did not change/add much code here, though it is reflecting on my changeset. I just changed the bundleElements to a Map variable.